Replace Paper with Vaadin Elements

manolo opened the pull request:
Connects to #358 

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-date-picker/372)
<!-- Reviewable:end -->

tomivirkki commented:
Review status: 0 of 17 files reviewed at latest revision, 8 unresolved discussions.

---

*[bower.json, line 25 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RhIvNCAccNuhXBQL:-Kl2RhIvNCAccNuhXBQM:bxfyfor) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L25)):*
> ```JSON
>     "iron-a11y-keys-behavior": "#2.0.0",
>     "iron-dropdown": "#2.0.0",
>     "iron-form-element-behavior": "#2.0.0",
> ```

Should be dropped

---

*[bower.json, line 31 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RckmclNJJDNDzdXM:-Kl2RckmclNJJDNDzdXN:bxfyfor) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L31)):*
> ```JSON
>     "iron-overlay-behavior": "#2.0.0",
>     "iron-resizable-behavior": "#2.0.0",
>     "iron-validatable-behavior": "#2.0.0",
> ```

Should be dropped

---

*[bower.json, line 35 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RSquj_wxHj8VK-wc:-Kl2RSquj_wxHj8VK-wd:b-oz40kw) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L35)):*
> ```JSON
>     "polymer": "^2.0",
>     "vaadin-button": "^1.0.0",
>     "vaadin-themable-mixin": "vaadin/vaadin-themable-mixin#^1.1.0",
> ```

Committed by accident?

---

*[bower.json, line 36 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RMYTSzBChMBLfEHJ:-Kl2RMYTSzBChMBLfEHK:b-oz40kw) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L36)):*
> ```JSON
>     "vaadin-button": "^1.0.0",
>     "vaadin-themable-mixin": "vaadin/vaadin-themable-mixin#^1.1.0",
>     "vaadin-themes": "vaadin/vaadin-themes#master",
> ```

Committed by accident?

---

*[bower.json, line 46 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2Rlxqre6pirn9Osjb:-Kl2Rlxqre6pirn9Osjc:b1ylaba) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L46)):*
> ```JSON
>     "iron-form": "#2.0-preview",
>     "iron-icons": "#2.0.0",
>     "iron-input": "#2.0.0",
> ```

Why iron-input dependency?

---

*[vaadin-date-picker-overlay.html, line 255 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SQNgSyzMPwPn3P4F:-Kl2SQNgSyzMPwPn3P4G:b-k6t24z) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L255)):*
> ```HTML
>           value=""
>           invalid="{{invalid}}"
>           placeholder="[[placeholder]]"
> ```

The `<div>` isn't probably supposed to have these properties

---

*[vaadin-date-picker-overlay.html, line 416 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SfvwP-NhZcdWv714:-Kl2SfvwP-NhZcdWv715:b17z7uw) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L416)):*
> ```HTML
>           super.connectedCallback();
>           this._translateX = this._yearScrollerWidth;
>           this.classList.add('animate');
> ```

Re-use `_toggleAnimateClass`

---

*[vaadin-date-picker-overlay.html, line 680 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SpsZfxmY24gEcbzX:-Kl2SpsZfxmY24gEcbzY:b-dlbqrr) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L680)):*
> ```HTML
>         _toggleAnimateClass(enable) {
>           if (enable) {
>             if (this.classList.contains('animate')) {
> ```

Unnecessary check

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Couple of requests to `.travis.yml`:
- Add `- polymer lint --rules polymer-2 --input *.html` to `before_script`
- The issues we had with hybrid version tests may no longer be there. Try switching back to running normal `polymer test` (instead of separate suite bundles)

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-Kl2VwFURJp0rerrfwXI:b-navp5y)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
We should probably re-enable iOS tests now that we're past hybrid. Current status is that they're not passing locally either.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-Kl2YW69uvh-u5VIY2LX:be8rl88)*
<!-- Sent from Reviewable.io -->

manolo commented:
> Add - polymer lint 

I already did, but it breaks the build since we have the late-upgrade test that uses a component without importing it.
No idea how to skip that check @limonte  ? 

>  re-enable iOS

Working on it in a separate branch in order not to break this PR build. I will merge when test passes if they are reliable because the problem was random failures in 

---

Review status: 0 of 19 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[bower.json, line 25 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RhIvNCAccNuhXBQL:-Kl2ZadFNL_9jeE1_7w_:bsyrmdt) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L25)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Should be dropped
</blockquote></details>

Removing it we break iron-form compatibility, there are 4 tests broken.

---

*[bower.json, line 31 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RckmclNJJDNDzdXM:-Kl2__nN3QKiU0dXLrwH:b-yx8fec) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L31)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Should be dropped
</blockquote></details>

Removing it we break validation tests.

I would remove legacy mixins in a separate PR. I don't think is needed right now for the pure P2, and we might wait until polymer releases pure E6 mixins for those.

---

*[bower.json, line 35 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RSquj_wxHj8VK-wc:-Kl2XraZEsJ4Keq4L0Qr:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L35)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Committed by accident?
</blockquote></details>

Done.

---

*[bower.json, line 36 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RMYTSzBChMBLfEHJ:-Kl2XImCVxnKj8FEJODZ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L36)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Committed by accident?
</blockquote></details>

Done.

---

*[bower.json, line 46 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2Rlxqre6pirn9Osjb:-Kl2XRT-CZWD26YflpmR:b-7yrp35) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L46)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why iron-input dependency?
</blockquote></details>

we use it in custom-input test and demos

---

*[vaadin-date-picker-overlay.html, line 255 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SQNgSyzMPwPn3P4F:-Kl2VI0y5QjYYdAj2Xre:b-k0c74l) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L255)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The `<div>` isn't probably supposed to have these properties
</blockquote></details>

Just left there by mistake because I just changed text-field by div and forgot revisit it. Removed.

---

*[vaadin-date-picker-overlay.html, line 416 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SfvwP-NhZcdWv714:-Kl2WCb9WAcFPLH-BSyv:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L416)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Re-use `_toggleAnimateClass`
</blockquote></details>

Done.

---

*[vaadin-date-picker-overlay.html, line 680 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SpsZfxmY24gEcbzX:-Kl2WL7vGh1ri3F_-g79:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L680)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Unnecessary check
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-Kl2WPnq99ZRJvRedCOj:be1r5yv)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 17 files reviewed at latest revision, 8 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 255 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2SQNgSyzMPwPn3P4F:-Kl2bFioY4RjEJzhSI-x:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/vaadin-date-picker-overlay.html#L255)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Just left there by mistake because I just changed text-field by div and forgot revisit it. Removed.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions.

---

*[bower.json, line 25 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RhIvNCAccNuhXBQL:-Kl2bbIsa9rXcAjHe8n9:bn653hq) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L25)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Removing it we break iron-form compatibility, there are 4 tests broken.
</blockquote></details>

That was the plan, right? https://reviewable.io/reviews/vaadin/vaadin-date-picker/370#-KkpFyPvTG3UehlBpgr2

---

*[bower.json, line 31 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RckmclNJJDNDzdXM:-Kl2bmseFPeBEoi5zG1o:bn653hq) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L31)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Removing it we break validation tests.

I would remove legacy mixins in a separate PR. I don't think is needed right now for the pure P2, and we might wait until polymer releases pure E6 mixins for those.
</blockquote></details>

That was the plan, right? https://reviewable.io/reviews/vaadin/vaadin-date-picker/370#-KkpFyPvTG3UehlBpgr2

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/form-input.html, line 97 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl3EXUaUYdFO1smvp2D:-Kl3EXUaUYdFO1smvp2E:bsxxr57) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/6516797f4c65f2921459cd3f979e76271ab54693/test/form-input.html#L97)):*
> ```HTML
>       });
> 
>       // TODO: support custom validatos somehow in datepicker
> ```

My proposal is to use a similar approach what `vaadin-text-field` uses. Adding a `checkValidity()` function which can be overridden by the user. By default, it would validate that the value in the input is a date. `validate()` function would assign the `invalid` property based on the return value of `checkValidity()`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/form-input.html, line 97 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl3EXUaUYdFO1smvp2D:-Kl3FzrBE5WMh2GBD9cG:b-m8yf2v) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/6516797f4c65f2921459cd3f979e76271ab54693/test/form-input.html#L97)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

My proposal is to use a similar approach what `vaadin-text-field` uses. Adding a `checkValidity()` function which can be overridden by the user. By default, it would validate that the value in the input is a date. `validate()` function would assign the `invalid` property based on the return value of `checkValidity()`
</blockquote></details>

That means to remove support for IronMeta based validators right ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

limonte commented:
>  I already did, but it breaks the build since we have the late-upgrade test that uses a component without importing it.
No idea how to skip that check @limonte ?

I fixed all polymer-lint issues except one, take a look - it look like the real issue with using property which wasn't defined.   

---

Review status: 0 of 19 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-Kl3QsLbbmsYlKPOzBye:b-hizh7e)*
<!-- Sent from Reviewable.io -->

limonte commented:
Btw, the current CI configuration looks bad, please consider reverting to the original configuration which was before Polymer2 things.

---

Review status: 0 of 19 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-Kl3W7SVeFuGeSGbbZH1:b-6g8tx3)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 22 files reviewed at latest revision, 3 unresolved discussions.

---

*[bower.json, line 25 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RhIvNCAccNuhXBQL:-Kl4fdCWVgUIkzJsqECD:ba3dfob) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L25)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

That was the plan, right? https://reviewable.io/reviews/vaadin/vaadin-date-picker/370#-KkpFyPvTG3UehlBpgr2
</blockquote></details>

Removed

---

*[bower.json, line 31 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl2RckmclNJJDNDzdXM:-Kl4fkU6A8Isbr9Vpoe7:b-l4tg4l) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/b3df9ba59ad8d9fefe1974886b44a471db7e2135/bower.json#L31)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

That was the plan, right? https://reviewable.io/reviews/vaadin/vaadin-date-picker/370#-KkpFyPvTG3UehlBpgr2
</blockquote></details>

Change the validation API so as we don't depend on validatable and the user can extend the element overriding the `checkValidity` method, so we are aligned with `vaadin-text-field`.  The component still is compatible with `iron-input` but we don't support `IronMeta` validators . Already discussed with @saulis 

---

*[test/form-input.html, line 97 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-Kl3EXUaUYdFO1smvp2D:-Kl4gIfrU_wY_rW4po-x:bruwhdu) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/6516797f4c65f2921459cd3f979e76271ab54693/test/form-input.html#L97)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

That means to remove support for IronMeta based validators right ? 
</blockquote></details>

Remved `IronMeta`, and implemented validations in the same way than `vaadin-text-field`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-date-picker-overlay.html, line 322 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI66S5I-7-_WQ8DBEK:-KlI66S5I-7-_WQ8DBEL:b-1q4az8) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L322)):*
> ```HTML
> 
> 
>       <vaadin-button id="todayButton" disabled="[[!_isTodayAllowed(minDate, maxDate)]]" on-tap="_onTodayTap">
> ```

For some reason the buttons are not accessible with keyboard (enter/space)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 22 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 23 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI72BGUIt_WonX2h0_:-KlI72BHGzUo7ak9VSeJ:b-jv2ddx) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L23)):*
> ```HTML
>   @font-face {
>     font-family: 'vaadin-date-picker-icons';
>     src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATUAAsAAAAABIgAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABPUy8yAAABCAAAAGAAAABgDxIFrGNtYXAAAAFoAAAAVAAAAFQXVtKJZ2FzcAAAAbwAAAAIAAAACAAAABBnbHlmAAABxAAAAPwAAAD8CiW322hlYWQAAALAAAAANgAAADYNObc8aGhlYQAAAvgAAAAkAAAAJAdCA8hobXR4AAADHAAAABwAAAAcEgACxGxvY2EAAAM4AAAAEAAAABAAcgDCbWF4cAAAA0gAAAAgAAAAIAALACNuYW1lAAADaAAAAUoAAAFKPg+r7HBvc3QAAAS0AAAAIAAAACAAAwAAAAMDgAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAA6QIDwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADgAAAAKAAgAAgACAAEAIOkC//3//wAAAAAAIOkA//3//wAB/+MXBAADAAEAAAAAAAAAAAAAAAEAAf//AA8AAQAAAAAAAAAAAAIAADc5AQAAAAABAAAAAAAAAAAAAgAANzkBAAAAAAEAAAAAAAAAAAACAAA3OQEAAAAAAQFvAMACqwLAAAUAAAEHFwcXAQGrPMPDPAEAAsA8xMQ8AQAAAQDVAJUDKwLrAAsAAAEnBycHFwcXNxc3JwMrPO/vPO/vPO/vPO8Crzzv7zzv7zzv7zzvAAMAgABAA4ADlQADABwAIAAAASMVMwMVITUjFSMiBhURFBYzITI2NRE0JisBNSMTIREhAtXV1Sr+qlUrIzIyIwJWIzIyIytVgP2qAlYBwNUCqlVVVTIj/aojMjIjAlYjMlX9AAHWAAAAAQAAAAEAABMCJlVfDzz1AAsEAAAAAADVTblVAAAAANVNuVUAAAAAA4ADlQAAAAgAAgAAAAAAAAABAAADwP/AAAAEAAAAAAADgAABAAAAAAAAAAAAAAAAAAAABwQAAAAAAAAAAAAAAAIAAAAEAAFvBAAA1QQAAIAAAAAAAAoAFAAeADAASgB+AAEAAAAHACEAAwAAAAAAAgAAAAAAAAAAAAAAAAAAAAAAAAAOAK4AAQAAAAAAAQACAAAAAQAAAAAAAgAHADMAAQAAAAAAAwACACcAAQAAAAAABAACAEgAAQAAAAAABQALAAYAAQAAAAAABgACAC0AAQAAAAAACgAaAE4AAwABBAkAAQAEAAIAAwABBAkAAgAOADoAAwABBAkAAwAEACkAAwABBAkABAAEAEoAAwABBAkABQAWABEAAwABBAkABgAEAC8AAwABBAkACgA0AGhkcABkAHBWZXJzaW9uIDEuMABWAGUAcgBzAGkAbwBuACAAMQAuADBkcABkAHBkcABkAHBSZWd1bGFyAFIAZQBnAHUAbABhAHJkcABkAHBGb250IGdlbmVyYXRlZCBieSBJY29Nb29uLgBGAG8AbgB0ACAAZwBlAG4AZQByAGEAdABlAGQAIABiAHkAIABJAGMAbwBNAG8AbwBuAC4AAAADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA) format('woff');
> ```

Still wondering if this could not be a external file `vaadin-date-picker-icons.woff` in case the user would like to modify this by CSS he can by changing the src property with a different url (external or data)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-date-picker-overlay.html, line 23 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI72BGUIt_WonX2h0_:-KlI7mfx6WF600rTJHPe:bgfkxtm) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L23)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Still wondering if this could not be a external file `vaadin-date-picker-icons.woff` in case the user would like to modify this by CSS he can by changing the src property with a different url (external or data)
</blockquote></details>

Either way works for me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 22 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 23 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI72BGUIt_WonX2h0_:-KlI832bpl-A4NLoKull:b5eft80) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L23)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Either way works for me.
</blockquote></details>

So, I think for us is more mainteinable to have the binary file, so as we can modify or track modifications easily. Also the code in this file is more readable without long lines. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

Saulis commented:
There's still the issue on Chrome where scrolling the page when overlay is open will also scroll the date-picker. Is this something we should fix before merging?

---

Review status: 0 of 22 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 282 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI-wxcUQIio3iP466p:-KlI-wxcUQIio3iP466q:b76d7e9) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L282)):*
> ```HTML
>     <div id="header" on-touchend="_preventDefault" aria-hidden="true">
>       <div id="label">[[label]]</div>
>       <div id="input" invalid="{{invalid}}">[[_formatDisplayed(selectedDate, i18n.formatDate)]]</div>
> ```

No reason for two-way binding here, `invalid$="[[invalid]]"` 
Also, can't find any styles defined for the `invalid` attribute? 

---

*[vaadin-date-picker-overlay.html, line 412 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI-n8C5QJlTzQnsutb:-KlI-n8GtH6aBS_YB7bx:b-wphi6g) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L412)):*
> ```HTML
>              * This property is set to true when the control value invalid.
>              */
>             invalid: {
> ```

Can't find a place where overlay `invalid` would be set?

---

*[test/form-input.html, line 106 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI8VIH4smXV3YR2w3e:-KlI8VIH4smXV3YR2w3f:b-mvdrgp) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/test/form-input.html#L106)):*
> ```HTML
> 
>         it('should validate correctly with custom validator', () => {
>           window.parent.e = datepicker;
> ```

Some debug line here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlI8lczkzzNA0JY8zhs:b95vn3f)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-date-picker-overlay.html, line 23 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI72BGUIt_WonX2h0_:-KlIB1ZPdTJ9PQghGGO5:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L23)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

So, I think for us is more mainteinable to have the binary file, so as we can modify or track modifications easily. Also the code in this file is more readable without long lines. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

manolo commented:
> There's still the issue on Chrome where scrolling

Added a workaround so as scrolling is disabled when the mouse outs.

---

Review status: 0 of 24 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 282 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI-wxcUQIio3iP466p:-KlIT1nIf0ROsJM112bz:ba3dfob) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L282)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

No reason for two-way binding here, `invalid$="[[invalid]]"` 
Also, can't find any styles defined for the `invalid` attribute? 
</blockquote></details>

Removed

---

*[vaadin-date-picker-overlay.html, line 322 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI66S5I-7-_WQ8DBEK:-KlIS-s2lAgK5UxPO_yu:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L322)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

For some reason the buttons are not accessible with keyboard (enter/space)
</blockquote></details>

Done.

---

*[vaadin-date-picker-overlay.html, line 412 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI-n8C5QJlTzQnsutb:-KlISZ_mpGETHjk6qoFP:b-354x0z) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L412)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Can't find a place where overlay `invalid` would be set?
</blockquote></details>

Removed, we don't need this  in the overlay. 

---

*[test/form-input.html, line 106 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI8VIH4smXV3YR2w3e:-KlISw7sZHo_lmEHwS_b:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/test/form-input.html#L106)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Some debug line here?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlIT8lZA4xWS-AbPO09:b-buwiwj)*
<!-- Sent from Reviewable.io -->

manolo commented:
@saulis removed the fix for the scrolling, since the issue is also in master and 1.x.
My fix needed more work, and possiblly causes a scroll sync regresion.
We should face this issue in a separate PR

---

Review status: 0 of 23 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlIfu4XirUQSo7_6nGU:b-slvjva)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-date-picker-overlay.html, line 322 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI66S5I-7-_WQ8DBEK:-KlJNXuOLAYjZuN2f6EQ:b-h9m828) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L322)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Still something strange with the Today-button. If you use it with enter, it shouldn't select the date but scroll to it (unless it's already scrolled to). This seems to work with click ok.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: 0 of 23 files reviewed at latest revision, 5 unresolved discussions.

---

*[bower.json, line 28 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlJHH9nnaFMz1m01pQU:-KlJHH9okW9FgZRAibqv:bvl1uvd) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/71fdce606d174898c32c2fad5302e6b2b5d5bd81/bower.json#L28)):*
> ```JSON
>     "iron-overlay-behavior": "#2.0.0",
>     "iron-resizable-behavior": "#2.0.0",
>     "paper-styles": "#2.0.0",
> ```

Do we still need `<paper-styles>`? Why?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[bower.json, line 28 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlJHH9nnaFMz1m01pQU:-KlJOhxUb8-Z-W_lM_Ha:bgrjlyu) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/71fdce606d174898c32c2fad5302e6b2b5d5bd81/bower.json#L28)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Do we still need `<paper-styles>`? Why?
</blockquote></details>

Will be removed as part of #359 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-date-picker-overlay.html, line 820 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlN8ak9o5tbocen1fEr:-KlN8ak9o5tbocen1fEs:b-f3kb8l) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/71fdce606d174898c32c2fad5302e6b2b5d5bd81/vaadin-date-picker-overlay.html#L820)):*
> ```HTML
>             }
>           } else if ((eventKey == 'enter' || eventKey == 'space') && this._buttonFocused()) {
>             // Let buttons handle their default actions
> ```

Event propagation should be stopped here to avoid date picker from selecting a value on enter.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: 0 of 23 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[bower.json, line 28 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlJHH9nnaFMz1m01pQU:-KlNJrhEK43XNGXpNPC1:bb6ywmx) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/71fdce606d174898c32c2fad5302e6b2b5d5bd81/bower.json#L28)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Will be removed as part of #359 
</blockquote></details>

LGTM

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Encountered a small regression with validation: If the component has a value and you type "foobar" into the field and hit enter, after leaving the field it appears as if "foobar" is a valid value for the field (while it's not the actual value).

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlNXYw0TNVEcul-CdMJ:bo5y4fe)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 23 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-date-picker-overlay.html, line 322 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlI66S5I-7-_WQ8DBEK:-KlNY9_4Qa3SjyojLdLZ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd9ed34487a6dbb9e6cff064d38fbc34bd73168f/vaadin-date-picker-overlay.html#L322)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Still something strange with the Today-button. If you use it with enter, it shouldn't select the date but scroll to it (unless it's already scrolled to). This seems to work with click ok.
</blockquote></details>

Done.

---

*[vaadin-date-picker-overlay.html, line 820 at r7](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-KlN8ak9o5tbocen1fEr:-KlNY8P13K7Ig5abbjSw:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/71fdce606d174898c32c2fad5302e6b2b5d5bd81/vaadin-date-picker-overlay.html#L820)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Event propagation should be stopped here to avoid date picker from selecting a value on enter.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372)*
<!-- Sent from Reviewable.io -->

manolo commented:
It's fixed now. 

---

Review status: 0 of 23 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlO86tZu2bjXhIwNIks:b-24bgjs)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Umm...it's quite broken now actually. The component turns invalid immediately when you start navigating the dates. And after picking a date it turns invalid when you leave the field even if it has a valid value.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlOEgvbqNhUYK9S2ad8:bbq1fa9)*
<!-- Sent from Reviewable.io -->

manolo commented:
Yes, making the input run datepicker validation meant that validation was run always the user moves the cursor on the overlay because changes in the focused day changes the input value.
I have fixed by disabling the validate method in the input.

---

Review status: 0 of 23 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/372#-:-KlOUcIVQjAebXNS0OzZ:br1eflt)*
<!-- Sent from Reviewable.io -->
